From: =?utf-8?q?S=C3=A9bastien_Villemot?= <sebastien@debian.org>
Date: Wed, 25 Aug 2021 00:00:00 +0000
Subject: Skip some tests when run with autopkgtest
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

- elfcore.test.sh and relocation.test.sh need to rebuild either libsbcl.a or
  shrinkwrap-sbcl, which can’t easily be done within autopkgtest.

- run-sbcl.test.sh does not make sense in an autopkgtest context, since its
  purpose is to test a not-yet-installed sbcl binary.
---
 tests/elfcore.test.sh    | 2 +-
 tests/relocation.test.sh | 2 +-
 tests/run-sbcl.test.sh   | 5 +++++
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/tests/elfcore.test.sh b/tests/elfcore.test.sh
index 8fbf4f9..fef280f 100755
--- a/tests/elfcore.test.sh
+++ b/tests/elfcore.test.sh
@@ -22,7 +22,7 @@ run_sbcl <<EOF
  (exit :code 2) ; otherwise
 EOF
 status=$?
-if [ $status != 0 ]; then # test can't be executed
+if [ $status != 0 ] || [ -n "${AUTOPKGTEST_TMP:-}" ]; then # test can't be executed
     # we don't have a way to exit shell tests with "inapplicable" as the result
     exit $EXIT_TEST_WIN
 fi
diff --git a/tests/relocation.test.sh b/tests/relocation.test.sh
index e62d125..be87593 100755
--- a/tests/relocation.test.sh
+++ b/tests/relocation.test.sh
@@ -7,7 +7,7 @@
 #        so there must have been a regression in the heap relocator.
 data=`run_sbcl --eval '(progn #+linux(progn(princ "fakemap") #+64-bit(princ "_64")))' \
   --quit`
-if [ -z "$data" ]
+if [ -z "$data" ] || [ -n "${AUTOPKGTEST_TMP:-}" ]
 then
     # shell tests don't have a way of exiting as "not applicable"
     exit $EXIT_TEST_WIN
diff --git a/tests/run-sbcl.test.sh b/tests/run-sbcl.test.sh
index b10191a..d6e6f12 100644
--- a/tests/run-sbcl.test.sh
+++ b/tests/run-sbcl.test.sh
@@ -1,3 +1,8 @@
+if [ -n "${AUTOPKGTEST_TMP:-}" ]
+then
+    exit $EXIT_TEST_WIN
+fi
+
 . ./subr.sh
 
 run_sbcl --eval '(exit :code (or #+unix 0 2))'
